File: Carrito-de-Compras/03-Fase3-CSharp/Models/Cliente.cs
namespace CarritoCompras.Models;

public class Cliente
{
    public int IdCliente { get; private set; }
    public string Nombre { get; private set; }
    public string Correo { get; private set; }
    private string Contrasena { get; set; }

    public Cliente(int idCliente, string nombre, string correo, string contrasena)
    {
        IdCliente = idCliente;
        Nombre = nombre;
        Correo = correo;
        Contrasena = contrasena;
    }

    public bool ValidarCredenciales(string correo, string contrasena)
    {
        return Correo.Equals(correo, StringComparison.OrdinalIgnoreCase)
               && Contrasena == contrasena;
    }
}

File: Carrito-de-Compras/02-Fase2-POO-BlueJ/Proyecto-BlueJ/Cliente.java
public class Cliente
{
    private int idCliente;
    private String nombre;
    private String correo;
    private String contrasena;
    private Carrito carrito;

    public Cliente(int idCliente, String nombre, String correo, String contrasena)
    {
        this.idCliente = idCliente;
        this.nombre = nombre;
        this.correo = correo;
        this.contrasena = contrasena;
        this.carrito = new Carrito();
    }

    public boolean iniciarSesion(String correo, String contrasena)
    {
        return this.correo.equals(correo) && this.contrasena.equals(contrasena);
    }

    public void verCarrito()
    {
        carrito.mostrarCarrito();
    }

    public Carrito getCarrito()
    {
        return carrito;
    }

    public String getNombre()
    {
        return nombre;
    }

    public int getIdCliente()
    {
        return idCliente;
    }
}
